backend/routes/users.js

```
@@ -45,19 +46,33 @@ router.route("/login").post(async (req, res) => {
   });
 });
 
-router.route("/update/:id").post((req, res) => {
+router.route("/update/:id").post(async (req, res) => {
+    const salt = 10;
     User.findById(req.params.id)
-        .then((user) => {
-            user.username = req.body.username;
-            user.password = req.body.password;
-            user.email = req.body.email;
-            user.avi = Number(req.body.avi);
-            user.chatrooms = req.body.chatrooms;
-            user.bio = req.body.bio;
-
+        .then(async (user) => {
+          if(req.body.username) { user.username= req.body.username; }
+          if(req.body.password) {
+            await Bcrypt.compare(req.body.password, user.password, async (err, result) => {
+              if (err){
+                res.status(400).json("Error: Invalid Current Password");
+              }
+              user.password= await Bcrypt.hash(req.body.password, salt);
+            });
+            user.password= await Bcrypt.hash(req.body.password, salt);
+          }
+          if(req.body.email) { user.email= req.body.email; }
+          if(req.body.avi) { user.avi= req.body.avi; }
+          if(req.body.chatrooms) { user.chatrooms= req.body.chatrooms; }
+          if(req.body.bio) { user.bio= req.body.bio; }
```

If someone sends an incorrect update request from the front end, won't the data in the database be overwritten here? If any of the data is missing and should be there, you should throw an error